figma-shop/src/app/components/login-dialog/login-dialog.component.html

```diff
@@ -0,0 +1,30 @@
+<form class="login-form" [formGroup]="loginForm" *ngIf="!userAuthStatus" (ngSubmit)="signIn()">
+  <h2>{{ signStatus ? 'Sign In With' : 'Sign Up' }}</h2>
+  <div class="social-login">
+    <div class="social social__facebook" (click)="loginVia('facebook')">
+      <img class="social__logo" src="./assets/images/facebook.png" alt="facebook">
+      <span>Facebook</span>
+    </div>
+    <div class="social social__google" (click)="loginVia('google')">
```

Comment:
You have to create `Enum` for all of this social names in separate file `social-type.enum.ts`, like this:
```ts
export enum SocialTypeEnum {
  facebook = 'facebook',
  google = 'google',
  ...
}
```
This is a bad practice to create enums in this way as I wrote, but you need a string and you may have an error if you write this enum in a proper way... The best way to write enums (in general) would be like this: 
```ts
export enum SocialTypeEnum {
  facebook = 1,
  google = 2,
  ...
}
```
You can find more information about enums here: 
![image](https://user-images.githubusercontent.com/17149365/103272880-a5444880-49c6-11eb-9cd5-f933feff11bd.png)

[Link here](https://github.com/labs42io/clean-code-typescript#dont-add-unneeded-context)

So this enum will help you not make an mistake...

_Example._
**Before:**
```ts
(click)="loginVia('google')"
```
**After:**
```ts
(click)="loginVia(SocialTypeEnum.google)"
```

Reply:
create enum